### PR TITLE
Week1 customer service

### PR DESCRIPTION
**Pull request by vikawal**
I've changed code, made components folder with files for each component and css style for them. My updated homework Priject e-commerce.

### CODE REVIEW

**Comment by vikawal** [comment]
Could deploy omly manually - https://bucolic-alfajores-4947ae.netlify.app/

File: week1/project/EcommerceVika/ecommerce-shop-project/src/App.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  return (
+    <div className="App">
+      <Categories
+        categories={allCategories}
+        selectedCategory={selectedCategory}
+        handleCategoryClick={handleCategoryClick}
+        />
+        <Products 
+        filteredProducts={filteredProducts}
+        />
```
Comment (ObadaMoustafa) [inline week1/project/EcommerceVika/ecommerce-shop-project/src/App.js]: indentation is very important .. at first I thought that `Products` is a child of `Categories` I do recommend you use an extension like [Pretier ](https://prettier.io/)of [ESlint ](https://eslint.org/) or if you prefer doing it manually but you need to take care about indentations

File: week1/project/EcommerceVika/ecommerce-shop-project/src/App.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React, { useState } from "react";
+import "./App.css";
+import Categories from "./components/Categories";
+import Products from "./components/Products";
+import FilteredProductsByCategory from "./components/FilteredProductsByCategory";
```
Comment (ObadaMoustafa) [inline week1/project/EcommerceVika/ecommerce-shop-project/src/App.js]: this is a normal function not a components .. it should use camelCase for its name

File: week1/project/EcommerceVika/ecommerce-shop-project/src/components/Categories.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+function Categories({categories, selectedCategory, handleCategoryClick}) {
+  
+  return (
+    <div className="categories">
+      
+        {categories.map((category) => (
+          <button
+            key={category}
+            onClick={() => handleCategoryClick(category)}
+            className={selectedCategory === category ? "active" : ""}
```
Comment (ObadaMoustafa) [inline week1/project/EcommerceVika/ecommerce-shop-project/src/components/Categories.js]: smart 💥

File: week1/project/EcommerceVika/ecommerce-shop-project/src/components/FilteredProductsByCategory.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+
+  const FilteredProductsByCategory = (products, category) => {
+  return category ? products.filter(
+        (product) => product.category === category.split(":")[1].trim()
+      )
+    : products;
+ };
+  
+export default FilteredProductsByCategory;
```
Comment (ObadaMoustafa) [inline week1/project/EcommerceVika/ecommerce-shop-project/src/components/FilteredProductsByCategory.js]: as I told in the previous comment this is a normal function not a component

File: week1/project/EcommerceVika/ecommerce-shop-project/src/components/Products.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            <div className="product-details">
+              <div className="product-title">{product.title}</div>
+              <div className="product-price">{product.price}</div>
+              <div className="product-description">{product.description}</div>
+              <div className="product-category">{product.category}</div>
+              <div className="product-rating">
+                Rating: {product.rating.rate} ({product.rating.count} reviews)
+              </div>
+            </div>
+          </div>
```
Comment (ObadaMoustafa) [inline week1/project/EcommerceVika/ecommerce-shop-project/src/components/Products.js]: this would be great if separated in a new component

**Review by ObadaMoustafa — COMMENTED** [review]
awesome and you have solved problems in a smart way. I am gonna approve it but you need to take care of the indentation .. I wrote some advice below but indentation is mandatory so please check all your files and fix it to the up coming homework

File: week1/project/EcommerceVika/ecommerce-shop-project/src/components/Categories.js (hunk 3 above)
Comment (vikawal) [reply week1/project/EcommerceVika/ecommerce-shop-project/src/components/Categories.js]: I saw it on stuckOverFlow... I steal... Is it normal???

Comment (ObadaMoustafa) [reply week1/project/EcommerceVika/ecommerce-shop-project/src/components/Categories.js]: the most important thing is you understand it .. while studying I recommend you never copy something you don't understand .. if you understand this code it's totally fine and solved your problem and gives you the result you want

File: week1/project/EcommerceVika/ecommerce-shop-project/src/App.js (hunk 1 above)
Comment (vikawal) [reply week1/project/EcommerceVika/ecommerce-shop-project/src/App.js]: I will read about indentation. It's how lines go after each other? Anna told me about this problem... but I'm so tired, when it works, that I usually forget((( I'll try to remember. Thank you for advise

File: week1/project/EcommerceVika/ecommerce-shop-project/src/App.js (hunk 2 above)
Comment (vikawal) [reply week1/project/EcommerceVika/ecommerce-shop-project/src/App.js]: So, components from BIG letter and functions - usual camelCase? Forgot, mixed. Sorry

File: week1/project/EcommerceVika/ecommerce-shop-project/src/components/FilteredProductsByCategory.js (hunk 4 above)
Comment (vikawal) [reply week1/project/EcommerceVika/ecommerce-shop-project/src/components/FilteredProductsByCategory.js]: I will make changes this night according to your comments. Thank you

File: week1/project/EcommerceVika/ecommerce-shop-project/src/components/Products.js (hunk 5 above)
Comment (vikawal) [reply week1/project/EcommerceVika/ecommerce-shop-project/src/components/Products.js]: You mean to make product.js with own css file? Right?

**Comment by vikawal** [comment]
I've cleaned code manually and added 1 component. Deployed: https://singular-boba-ed6d0e.netlify.app/

**Comment by ObadaMoustafa** [comment]
wonderful

**Pull request closed without merging**